assets/src/edit-story/components/library/panes/text/textPane.js

```diff
@@ -32,17 +33,67 @@ import { Section, MainButton, SearchInput } from '../../common';
 import { FontPreview } from '../../text';
 import useLibrary from '../../useLibrary';
 import { Pane } from '../shared';
+import { dataFontEm } from '../../../../units';
+import { useHistory } from '../../../../app/history';
 import paneId from './paneId';
 import { PRESETS, DEFAULT_PRESET } from './textPresets';
 
 const SectionContent = styled.p``;
 
+// The format is:
+// { presetToBeAddedName: [list of previous preset names that need to be considered]
+// If the previous preset is not in the list of names, no special rules.
+// If the preset being added is not in the rules at all -- no special rules.
+// @todo Text staggering to be done as part of https://github.com/google/web-stories-wp/issues/1206
+const POSITIONING_RULES = {
+  'heading-2': ['heading-1'],
+  'heading-3': ['heading-1', 'heading-2'],
+  paragraph: ['heading-1', 'heading-2', 'heading-3'],
+};
```

Comment:
Hm, while I agree, that this is one possible interpretation of the UX requirements, I'm not sure I think it makes sense. Is this useful in such a limited fashion?

Reply:
Did you understand the UX requirements in a different way? If yes, please do let me know :)

Note that staggering the same text element will be done in #1206 -- if I understand correctly, the specific issue seems to be for the consecutive different texts staggering only, I might be mistaken though, perhaps this issue should include generally all text staggering.

We could definitely bring this up tomorrow over the grooming to understand the UX goal better.